exec/vector/src/main/java/org/apache/drill/exec/record/metadata/MetadataUtils.java

```diff
@@ -187,6 +187,10 @@ public static ColumnMetadata newMapArray(String name, TupleMetadata schema) {
     return new MapColumnMetadata(name, DataMode.REPEATED, (TupleSchema) schema);
   }
 
+  public static DictColumnMetadata newDictArray(String name, TupleMetadata schema) {
+    return new DictColumnMetadata(name, DataMode.REPEATED, (TupleSchema) schema);
```

Comment:
Picking up on a comment in your PR description, I wonder if we have the wrong semantics here. It is true that DICT is *implemented* as a map. But, at the metadata (descriptive) level, it is not a map, and is not constructed from a map; it is instead a `<KEY,VALUE>` pair.

I'm not sure we're doing ourselves a favor by exposing the implementation detail (a kind of map) in our metadata description.

Reply:
Let me explain: Dict stores `key` and `value` `ColumnMetadata`s in `TupleSchema` - the way Map's members are stored - but there's a validation for each of the fields (name, type). Dict does not contain a Map, it stores its `key` and `value` in its `TupleSchema schema` field.

When `TupleMetadata` is constructed for Parquet table as part of table metadata, we loop over each Parquet field's `SchemaPath` representation (leaf fields, e.g. `` `mapcol`.`map`.`key` ``, `` `structcol`.`b` `` with the last field in schema being a primitive). Such named segments are treated as either a (Drill's) `MAP` or `DICT`, depending on parent segment type.

(Parquet's) `MAP` is represented as a group (note, that nested group's name, `key_value` below, can be different, based on the system which produced the Parquet file):
```
<map-repetition> group <name> (MAP) {
  repeated group key_value {
    required <key-type> key;
    <value-repetition> <value-type> value;
  }
}
```
and before changes in the PR, when `TupleMetadata` was being created for the table, if `DICT` column was encountered, it included a nested `key_value` group as Drill's `MAP` which then contained `key` and `value` fields. Thus, there is a need to skip this segment if we know that its parent's type is `DICT` to have correct `ColumnMetadata` for the `DICT` field.